### PR TITLE
feat(FN-3575): add initial record correction creation E2E tests

### PR DESCRIPTION
**Pull request by sw-mattw**
## Introduction :pencil2:
The initial part of the record correction creation journey requires E2E tests now the initial work for this has been merged.

## Resolution :heavy_check_mark:
- Add E2E tests for:
  - Checking the selected fee record summary is rendered correctly
  - Checking any creation form validation errors are rendered
  - Checking submission of the creation form with valid form values redirects the user to the "check the information" page

Note that the work to persist the form submission has not been completed yet and so the E2E tests for this work will follow in a subsequent PR.

Also note that this PR is based on the validation work (PR: https://github.com/UK-Export-Finance/dtfs2/pull/3948), this PR should not be merged until this base has been merged into main.

## Miscellaneous :heavy_plus_sign:
- Simplified the naming of the fee record summary component's `data-cy` attribute value

### CODE REVIEW

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 1]
```diff
@@ -67,13 +73,91 @@ context('When fee record correction feature flag is enabled', () => {
...
+        cy.assertText(mainHeading(), 'Record correction request');
+
+        feeRecordSummary.container().should('exist');
+        cy.assertText(feeRecordSummary.facilityId(), feeRecordAtToDoStatus.facilityId);
+        cy.assertText(feeRecordSummary.exporter(), feeRecordAtToDoStatus.exporter);
+        cy.assertText(feeRecordSummary.requestedBy(), `${USERS.PDC_RECONCILE.firstName} ${USERS.PDC_RECONCILE.lastName}`);
+      });
+    });
+
+    context('when the user submits the "create record correction request" form with invalid values', () => {
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:108]: I'm not familiar with the design/functionality - do we have/need a test for submitting an empty form?

File: e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/createFeeRecordCorrectionRequestPage.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const createFeeRecordCorrectionRequestPage = {
+  mainHeading: () => cy.get(`[data-cy="main-heading"]`),
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/createFeeRecordCorrectionRequestPage.js:2]: This should come from partials now to be DRY

**Review by ttbarnes — CHANGES_REQUESTED** [review]
2x minor comments/question please

File: e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/createFeeRecordCorrectionRequestPage.js (hunk 2 above)
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/pages/utilisation-reports/fee-record-correction/createFeeRecordCorrectionRequestPage.js]: Thanks, removed this and now using the TFM partial

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js (hunk 1 above)
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js]: I've refactored these tests a little to try and make things a bit simpler when we add more stages to the journey (upcoming tickets).

This has now been moved to the new error cases file, I've also added a test for an empty form as suggested - thanks!

Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js]: For a bit more context, the form validation requirements are:
- 1 or more reasons checked
- Between 1 and 500 characters of additional info

These are tested in unit tests, but I've now got e2e tests (in this PR) for: checking multiple reasons and having valid additional info input, no checked reasons and no additional info input, more additional info characters than allowed

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 3]
```diff
@@ -67,13 +73,91 @@ context('When fee record correction feature flag is enabled', () => {
...
+
+      it('should render the create record correction request page with selected fee record details', () => {
+        cy.assertText(mainHeading(), 'Record correction request');
+
+        feeRecordSummary.container().should('exist');
+        cy.assertText(feeRecordSummary.facilityId(), feeRecordAtToDoStatus.facilityId);
+        cy.assertText(feeRecordSummary.exporter(), feeRecordAtToDoStatus.exporter);
+        cy.assertText(feeRecordSummary.requestedBy(), `${USERS.PDC_RECONCILE.firstName} ${USERS.PDC_RECONCILE.lastName}`);
+      });
+    });
```
[the remark this answers is not in the document]
Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js]: Yep, thanks for suggesting - extended the existing test case instead

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 4]
```diff
@@ -49,34 +55,46 @@ context('When fee record correction feature flag is enabled', () => {
...
       premiumPaymentsTab.createRecordCorrectionRequestButton().click();
 
-      cy.url().should('eq', relative(`/utilisation-reports/${reportId}`));
+      cy.url().should('eq', relative(`/utilisation-reports/${reportId}/create-record-correction-request/${feeRecordAtToDoStatus.id}`));
 
-      errorSummary().contains('Select a record to create a record correction request');
-      premiumPaymentsTab.premiumPaymentsTable.error().should('exist');
-      cy.assertText(premiumPaymentsTab.premiumPaymentsTable.error(), 'Error: Select a record to create a record correction request');
-    });
+      //---------------------------------------------------------------
```
Comment (sw-mattw) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:75]: The diff above has got slightly confusing - the error test case that was here has been removed as this is a duplicate of a test case in the existing premium payments error file (added prior to this PR)

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 5]
```diff
@@ -49,34 +55,46 @@ context('When fee record correction feature flag is enabled', () => {
...
+      // Create record correction
```
Comment (RLavender98) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:76]: I think these comments could be slightly more descriptive, could go in a couple of directions, perhaps either:
- We go with page names but are more specific e.g. 'create record correction request page' and 'create record correction request - check the information page'
- Or we go more with what actions the user is taking e.g. 'Fill in record correction request details' and 'Checks the information entered for record correction is correct'

**Review by RLavender98 — DISMISSED** [review]
Nice changes, I think this makes for more extensible tests as we build out more of this feature thanks!

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js (hunk 4 above)
Comment (ttbarnes) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:75]: Do we need this line @sw-mattw or can it be moved, These two lines don't make sense:

```
Complete record correction request information form
```
- Text is then asserted rather than completing a form

```
Checks the information entered for record correction is correct
```
A URL is then asserted

Comment (sw-mattw) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:75]: I've changed this to be paged-based naming (vs action) so the comments should make more sense now thanks

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 6]
```diff
@@ -49,34 +55,46 @@ context('When fee record correction feature flag is enabled', () => {
...
 
-      cy.url().should('eq', relative(`/utilisation-reports/${reportId}`));
+      cy.url().should('eq', relative(`/utilisation-reports/${reportId}/create-record-correction-request/${feeRecordAtToDoStatus.id}`));
 
-      errorSummary().contains('Select a record to create a record correction request');
-      premiumPaymentsTab.premiumPaymentsTable.error().should('exist');
-      cy.assertText(premiumPaymentsTab.premiumPaymentsTable.error(), 'Error: Select a record to create a record correction request');
-    });
+      //---------------------------------------------------------------
+      // Create record correction request page
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:76]:
```suggestion
      // Check the "Create record correction request" page
```

Otherwise to me it reads like it's creating something

File: e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js [hunk 7]
```diff
@@ -49,34 +55,46 @@ context('When fee record correction feature flag is enabled', () => {
...
-      premiumPaymentsTab.createRecordCorrectionRequestButton().click();
+      createFeeRecordCorrectionRequestPage.reasonCheckbox(RECORD_CORRECTION_REASON.FACILITY_ID_INCORRECT).check();
+      createFeeRecordCorrectionRequestPage.reasonCheckbox(RECORD_CORRECTION_REASON.OTHER).check();
+      cy.keyboardInput(createFeeRecordCorrectionRequestPage.additionalInfoInput(), 'Some additional info.');
 
-      cy.url().should('eq', relative(`/utilisation-reports/${reportId}/create-record-correction-request/${feeRecordAtToDoStatus.id}`));
+      cy.clickContinueButton();
+
+      //---------------------------------------------------------------
+      // Create record correction request - "check the information" page
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:92]: I don't think we need this comment it's just asserting a URL

Comment (ttbarnes) [reply e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/fee-record-correction/feature-flag-enabled/create-fee-record-correction.spec.js:92]: Ideally should be in it's own test i.e

```js
it('should redirect to ...')
```

But appreciate this wasn't created in this PR so not a blocker

**Review by ttbarnes — APPROVED** [review]

**Review by RLavender98 — APPROVED** [review]
